Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -4219,6 +4219,32 @@ - (WKFullScreenWindowController *)fullScreenWindowController
     return _fullScreenWindowController.get();
 }
 
+#if PLATFORM(VISION)
+
+- (UIMenu *)fullScreenWindowSceneDimmingAction
+{
+    UIDeferredMenuElement *deferredMenu = [UIDeferredMenuElement elementWithUncachedProvider:[weakSelf = WeakObjCPtr<WKWebView>(self)] (void (^completion)(NSArray<UIMenuElement *> *)) {
+        auto strongSelf = weakSelf.get();
+        if (!strongSelf)
+            return;
```

Comment:
Do we still want to invoke the completion handler here, even if we early return?

Reply:
Good catch!